### PR TITLE
DM: Update GitHub URLs for DM

### PR DESCRIPTION
**Pull request by dveeden**
### What is changed, added or deleted? (Required)

https://github.com/pingcap/dm has been archived. DM has moved to https://github.com/pingcap/tiflow/tree/master/dm

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v8.0 (TiDB 8.0 versions)
- [ ] v7.6 (TiDB 7.6 versions)
- [x] v7.5 (TiDB 7.5 versions)
- [ ] v7.4 (TiDB 7.4 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from:
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tidb-troubleshooting-map.md [hunk 1]
```diff
@@ -472,7 +472,7 @@ Check the specific cause for busy by viewing the monitor **Grafana** -> **TiKV**
 
 ### 6.2 Data Migration
 
-- 6.2.1 TiDB Data Migration (DM) is a migration tool that supports data migration from MySQL/MariaDB into TiDB. For details, see [DM on GitHub](https://github.com/pingcap/dm/).
+- 6.2.1 TiDB Data Migration (DM) is a migration tool that supports data migration from MySQL/MariaDB into TiDB. For details, see [DM Overview](dm/dm-overview.md).
```
Comment (dveeden) [inline tidb-troubleshooting-map.md:475]:
```suggestion
- 6.2.1 TiDB Data Migration (DM) is a migration tool that supports data migration from MySQL/MariaDB into TiDB. For details, see [DM Overview](/dm/dm-overview.md).
```

File: dm/dm-faq.md [hunk 2]
```diff
@@ -25,7 +25,7 @@ Currently, DM does not support it and only supports the regular expressions of t
 
 ## If a statement executed upstream contains multiple DDL operations, does DM support such migration?
 
-DM will attempt to split a single statement containing multiple DDL change operations into multiple statements containing only one DDL operation, but might not cover all cases. It is recommended to include only one DDL operation in a statement executed upstream, or verify it in the test environment. If it is not supported, you can file an [issue](https://github.com/pingcap/dm/issues) to the DM repository.
+DM will attempt to split a single statement containing multiple DDL change operations into multiple statements containing only one DDL operation, but might not cover all cases. It is recommended to include only one DDL operation in a statement executed upstream, or verify it in the test environment. If it is not supported, you can file an [issue](https://github.com/pingcap/tiflow/issues) to the tiflow repository.
```
Comment (Oreoxmt) [inline dm/dm-faq.md:28]:
```suggestion
DM will attempt to split a single statement containing multiple DDL change operations into multiple statements containing only one DDL operation, but might not cover all cases. It is recommended to include only one DDL operation in a statement executed upstream, or verify it in the test environment. If it is not supported, you can file an [issue](https://github.com/pingcap/tiflow/issues) to the `pingcap/tiflow` repository.
```

File: dm/quick-start-with-dm.md [hunk 3]
```diff
@@ -6,7 +6,7 @@ aliases: ['/docs/tidb-data-migration/dev/get-started/']
 
 # Quick Start Guide for TiDB Data Migration
 
-This document describes how to migrate data from MySQL to TiDB using [TiDB Data Migration](https://github.com/pingcap/dm) (DM). This guide is a quick demo of DM features and is not recommended for any production environment.
+This document describes how to migrate data from MySQL to TiDB using [TiDB Data Migration](/dm/dm-overview.md) (DM). This guide is a quick demo of DM features and is not recommended for any production environment.
```
Comment (Oreoxmt) [inline dm/quick-start-with-dm.md:9]:
```suggestion
This document describes how to migrate data from MySQL to TiDB using [TiDB Data Migration (DM)](/dm/dm-overview.md). This guide is a quick demo of DM features and is not recommended for any production environment.
```

> If the text includes an abbreviation in parentheses, include the long form and the abbreviation in the link text.
https://developers.google.com/style/link-text

File: dm/shard-merge-best-practices.md [hunk 4]
```diff
@@ -6,7 +6,7 @@ aliases: ['/docs/tidb-data-migration/dev/shard-merge-best-practices/']
 
 # Best Practices of Data Migration in the Shard Merge Scenario
 
-This document describes the features and limitations of [TiDB Data Migration](https://github.com/pingcap/dm) (DM) in the shard merge scenario and provides a data migration best practice guide for your application (the default "pessimistic" mode is used).
+This document describes the features and limitations of [TiDB Data Migration](/dm/dm-overview.md) (DM) in the shard merge scenario and provides a data migration best practice guide for your application (the default "pessimistic" mode is used).
```
Comment (Oreoxmt) [inline dm/shard-merge-best-practices.md:9]:
```suggestion
This document describes the features and limitations of [TiDB Data Migration (DM)](/dm/dm-overview.md) in the shard merge scenario and provides a data migration best practice guide for your application (the default "pessimistic" mode is used).
```

File: tidb-troubleshooting-map.md [hunk 5]
```diff
@@ -472,7 +472,7 @@ Check the specific cause for busy by viewing the monitor **Grafana** -> **TiKV**
 
 ### 6.2 Data Migration
 
-- 6.2.1 TiDB Data Migration (DM) is a migration tool that supports data migration from MySQL/MariaDB into TiDB. For details, see [DM on GitHub](https://github.com/pingcap/dm/).
+- 6.2.1 TiDB Data Migration (DM) is a migration tool that supports data migration from MySQL/MariaDB into TiDB. For details, see [DM Overview](/dm/dm-overview.md).
```
Comment (Oreoxmt) [inline tidb-troubleshooting-map.md:475]:
```suggestion
- 6.2.1 TiDB Data Migration (DM) is a migration tool that supports data migration from MySQL/MariaDB into TiDB. For details, see [DM overview](/dm/dm-overview.md).
```

**Review by Oreoxmt — APPROVED** [review]
Rest LGTM

**Review by lance6716 — APPROVED** [review]

**Pull request merged**